ibm-messaging-mq-cloud-showcase-app/backend/AWS-showcase-backend/models/Responder.js

```diff
@@ -43,7 +43,9 @@ class Responder {
             })
             .catch((err) => {
                 debug_warn(`Responder ${this.myID} error obtaining message ${err}`);
-                reject();
+                debug_warn("Potential Poison Message scenario encountered, invoking the poison message handler");
+                this.mqclient.poisonMessageHandler();
```

Comment:
poison message handling here, will only be for messages that can't be read. Messages without valid reply to queues will not fall into this catch.